File: trunk/original/wavelets/psijni/jnixform/stackarray.cpp
#include "stackarray.hpp"

        
stackarray::stackarray(int maxsize)
{
    size = maxsize;
    s = new quadnode [maxsize];
    N = 0;
}

int stackarray::empty () const
{
    return N == 0;
}

void stackarray::push (quadnode &item)
{
    //s[N++] = item;
    s[N++].copy (item);
}

void stackarray::pop (quadnode &item)
{
    item.copy ( s[--N]);
}

File: trunk/original/wavelets/psijni/jnixform/stackarray.java


class stackarray 
{
    private quadnode s[];
    private int N, size;
    public stackarray (int maxsize)
    {
        size = maxsize;
        s = new quadnode [maxsize];
        N = 0;
    }

    boolean empty () 
    {
        if ( N == 0) return true;
        return false;
    }   
    void push (quadnode item)
    {
        s[N++] = item;
    }
    quadnode pop ()
    {
        return s[--N];
    }
    
}